src/Sentry.AspNetCore/SentryHostBuilderExtensions.cs

```diff
@@ -0,0 +1,107 @@
+            => builder.UseSentry(o => o.Dsn = dsn);
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureOptions">The configure options.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((_, options) => configureOptions?.Invoke(options));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureOptions">The configure options.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, SentryAspNetCoreOptions>? configureOptions)
+            => builder.UseSentry((context, sentryBuilder) =>
+                sentryBuilder.AddSentryOptions(options => configureOptions?.Invoke(context, options)));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<ISentryBuilder>? configureSentry) =>
+            builder.UseSentry((_, sentryBuilder) => configureSentry?.Invoke(sentryBuilder));
+
+        /// <summary>
+        /// Uses Sentry integration.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="configureSentry">The Sentry builder.</param>
+        /// <returns></returns>
+        public static IHostBuilder UseSentry(
+            this IHostBuilder builder,
+            Action<HostBuilderContext, ISentryBuilder>? configureSentry)
+        {
+            // The earliest we can hook the SDK initialization code with the framework
+            // Initialization happens at a later time depending if the default MEL backend is enabled or not.
+            // In case the logging backend was replaced, init happens later, at the StartupFilter
+            _ = builder.ConfigureLogging((context, logging) =>
+            {
+                logging.AddConfiguration();
+
+                var section = context.Configuration.GetSection("Sentry");
+                _ = logging.Services.Configure<SentryAspNetCoreOptions>(section);
+
+                _ = logging.Services
+                    .AddSingleton<IConfigureOptions<SentryAspNetCoreOptions>, SentryAspNetCoreOptionsSetup>();
+                _ = logging.Services.AddSingleton<ILoggerProvider, SentryAspNetCoreLoggerProvider>();
+
+                _ = logging.AddFilter<SentryAspNetCoreLoggerProvider>(
```

Comment:
There is no TryAddFilter, i'm not sure it is an issue if this is added twice thought

Reply:
Not sure either, I guess something we'd need to test?  There's no way to look into the filters already added?

Reply:
Looks like it boils down to: https://github.com/aspnet/Logging/blob/2d2f31968229eddb57b6ba3d34696ef366a6c71b/src/Microsoft.Extensions.Logging/FilterLoggingBuilderExtensions.cs#L75